File: channel/rest/src/main/java/com/serious/service/channel/dispatch/ServiceRequest.java
package com.serious.service.channel.dispatch;
/*
 * @COPYRIGHT (C) 2016 Andreas Ernst
 *
 * All rights reserved
 */

import java.io.Serializable;

public class ServiceRequest implements Serializable {
    // instance data
    public String service;
    public int method;
    public Object[] arguments;

    // constructor

    ServiceRequest( String service, int method, Object[] arguments) {
        this.service   = service;
        this.method    = method;
        this.arguments = arguments;
    }
}

File: channel/rest/src/main/java/com/serious/service/channel/dispatch/ServiceRequest.kt
package com.serious.service.channel.dispatch
/*
* @COPYRIGHT (C) 2016 Andreas Ernst
*
* All rights reserved
*/

import java.io.Serializable

/**
 * A <code>ServiceRequest</code> covers all attributes required to call a service methods.
 */
data class ServiceRequest(
    var service: String,
    var method: Int,
    var arguments: Array<Any>
) :Serializable